src/fpm/manifest/preprocess.f90

```diff
@@ -228,4 +237,71 @@ logical function preprocess_is_same(this,that)
 
     end function preprocess_is_same
 
+    !> Clean preprocessor structure
+    elemental subroutine destroy(this)
+       class(preprocess_config_t), intent(inout) :: this
+
+       if (allocated(this%name))deallocate(this%name)
+       if (allocated(this%suffixes))deallocate(this%suffixes)
+       if (allocated(this%directories))deallocate(this%directories)
+       if (allocated(this%macros))deallocate(this%macros)
+
+    end subroutine destroy
+
+    !> Add preprocessor settings
+    subroutine add_config(this,that)
+       class(preprocess_config_t), intent(inout) :: this
+        type(preprocess_config_t), intent(in) :: that
+
+        if (.not.that%is_cpp()) then
+            write(stderr, '(a)') 'Warning: Preprocessor ' // that%name // &
+                                 ' is not supported; will ignore it'
+            return
+        end if
```

Comment:
This seems not to allow for the use of `fypp`. Is that intentional? Other code added seems to at least acknowledge its existence.

Reply:
No, indeed, `fypp` is not fully supported. However, this allows to use a [wrapper script](https://github.com/fortran-lang/fpm/pull/729/files#diff-0856223a01891eed802fbffe8ccc20b27618687ab837e66a7c714804d762198e) as proposed by @arteevraina .

With this PR #982 and @arteevraina wrapper script, we can compile `stdlib` from scratch with `fpm`. I guess that supporting fully `fypp` would come in a next PR.

Reply:
Note that this will allow to use suffixes other than `.f90` or `.F90` with `cpp` too.

Reply:
Ok , fair enough.

Reply:
Yep, this PR is limited to parsing user requests for custom suffixes - it does not address preprocessor handling which is unchanged (and like you're saying, `fypp` is not fully supported yet)